### PR TITLE
Add node e2e tests for hostNetwork

### PR DESCRIPTION
**Pull request by feiskyer**
**What this PR does / why we need it**:

Add node e2e tests for hostNetwork.

**Which issue this PR fixes**

Part of #44118.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

/assign @Random-Liu @yujuhong

### CODE REVIEW

**Comment by feiskyer** [comment]
@k8s-bot kubemark e2e test this

File: test/e2e_node/security_context_test.go [hunk 1]
```diff
@@ -114,4 +115,75 @@ var _ = framework.KubeDescribe("Security Context", func() {
...
+						{
+							Image:   image,
+							Name:    podName,
+							Command: command,
+						},
+					},
+				},
+			}
+		}
+		listNetworkInterfaceCommand := []string{"sh", "-c", "cat /proc/net/dev | awk '{print $1}'"}
```
Comment (php-coder) [inline test/e2e_node/security_context_test.go]: `cat /proc/net/dev | awk '{print $1}'` can be simplified into `awk '{print $1}' /proc/net/dev`

Comment (feiskyer) [inline test/e2e_node/security_context_test.go]: Good catch. Updated.

File: test/e2e_node/security_context_test.go [hunk 2]
```diff
@@ -114,4 +115,75 @@ var _ = framework.KubeDescribe("Security Context", func() {
...
+		BeforeEach(func() {
+			output, err := exec.Command(listNetworkInterfaceCommand[0], listNetworkInterfaceCommand[1:]...).Output()
+			if err != nil {
+				framework.Failf("Execute command %q failed: %v", listNetworkInterfaceCommand, err)
+			}
+			networkInterfaces = strings.TrimSpace(string(output))
+			framework.Logf("Got host network interfaces %q", networkInterfaces)
+		})
+
+		It("should show same network interfaces in the host network containers", func() {
```
Comment (yujuhong) [inline test/e2e_node/security_context_test.go]: It's possible that other pods were created after `BeforeEach` but before creating this test pod. If that's the case, won't the test pod see more interfaces? 

Besides running this in a serial suite, are there more robust way to check the host network setting?

File: test/e2e_node/security_context_test.go [hunk 3]
```diff
@@ -114,4 +115,75 @@ var _ = framework.KubeDescribe("Security Context", func() {
...
+						{
+							Image:   image,
+							Name:    podName,
+							Command: command,
+						},
+					},
+				},
+			}
+		}
+		listNetworkInterfaceCommand := []string{"sh", "-c", "awk '{print $1}' /proc/net/dev"}
```
Comment (yujuhong) [inline test/e2e_node/security_context_test.go]: Can you use `/sys/class/net` instead?

**Review by yujuhong — COMMENTED** [review]
@freehan do you have better ideas for verifying the host network setting?

**Comment by feiskyer** [comment]
@yujuhong Updated. PTAL.

**Comment by feiskyer** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by feiskyer** [comment]
ping @yujuhong @freehan @Random-Liu

File: test/e2e_node/security_context_test.go [hunk 4]
```diff
@@ -195,4 +196,80 @@ var _ = framework.KubeDescribe("Security Context", func() {
...
+
+		It("should listen on same port in the host network containers", func() {
+			busyboxPodName := "busybox-hostnetwork-" + string(uuid.NewUUID())
+			createAndWaitHostNetworkPod(busyboxPodName, true)
+			logs, err := framework.GetPodLogs(f.ClientSet, f.Namespace.Name, busyboxPodName, busyboxPodName)
+			if err != nil {
+				framework.Failf("GetPodLogs for pod %q failed: %v", busyboxPodName, err)
+			}
+
+			if !strings.Contains(logs, listeningPort) {
```
Comment (freehan) [inline test/e2e_node/security_context_test.go]: You probably want to print the logs?

Otherwise, there is no trace. The same for the next test.

**Comment by freehan** [comment]
Just nits. LGTM overall

File: test/e2e_node/security_context_test.go (hunk 4 above)
Comment (feiskyer) [inline test/e2e_node/security_context_test.go]: ACK

**Comment by feiskyer** [comment]
@freehan Updated. PTAL.

**Comment by freehan** [comment]
Thanks
/lgtm

**Comment by feiskyer** [comment]
ping @Random-Liu

**Comment by feiskyer** [comment]
ping @Random-Liu

**Comment by feiskyer** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Pull request merged**